src/thread.rs

```diff
@@ -0,0 +1,66 @@
+use std::fmt::{self, Debug};
+
+#[cfg(syn_can_use_thread_id)]
+mod imp {
+    use std::thread::{self, ThreadId};
+
+    /// ThreadBound is a Sync-maker and Send-maker that allows accessing a value
+    /// of type T only from the original thread on which the ThreadBound was
+    /// constructed.
+    pub struct ThreadBound<T> {
```

Comment:
I wonder if this should be split out into a separate crate rather than used internally - but *shrug*. It's not super generally useful & that might have some build overhead.

Reply:
Might as well make a crate for it. I can take care of that this week. I will check what the impact on build time turns out to be but I am guessing it will be noticeable enough that we shouldn't use the new crate in Syn.